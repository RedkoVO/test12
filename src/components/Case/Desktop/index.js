import React from 'react'
import PropTypes from 'prop-types'
import withStyles from '@material-ui/core/styles/withStyles'

import MenuNavigation from '../../MenuNavigation'
import Header from '../../../containers/Header/AsyncHeader'
import Footer from '../../Footer/Desktop/'
import CaseFilter from './components/CaseFilter'
import CaseOpen from './components/CaseOpen'
import CasesRelated from './components/CasesRelated'

import styles from './styles'

const Case = ({ classes, handleLogout, caseItems, relaredItems }) => (
  <div className="mainWrapperTrade">
    <div className={classes.containerTradeWrap}>
      <div className={classes.containerTrade}>
        <header className={classes.headerCase}>
          <MenuNavigation page="dark" handleLogout={handleLogout} />
          <Header type="catalog" />
        </header>
        <main className={classes.case}>
          {/* TODO: Similar like Cases component */}
          <CaseFilter />
          <CaseOpen caseItems={caseItems} />
          <CasesRelated relaredItems={relaredItems} />
        </main>
      </div>
    </div>
    <Footer />
  </div>
)

Case.propTypes = {
  classes: PropTypes.object,
  caseItems: PropTypes.array,
  relaredItems: PropTypes.array
}

export default withStyles(styles)(Case)
