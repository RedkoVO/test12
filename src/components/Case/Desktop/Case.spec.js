import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import Case from './'

import GunCase from '../../../assets/images/GunCase.png'
import CaseX from '../../../assets/images/CaseX.png'

const caseItems = [
  { id: 1, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 2, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 3, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 4, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 5, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 6, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 7, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 8, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 9, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 10, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 11, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 12, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 13, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 14, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 15, title: 'Good Weapon2 1.2', img: GunCase },
  { id: 16, title: 'Good Weapon2 1.2', img: GunCase }
]
const relaredItems = [
  { id: 1, title: 'Good Weapon2 1.2', img: CaseX },
  { id: 2, title: 'Good Weapon2 1.2', img: CaseX },
  { id: 3, title: 'Good Weapon2 1.2', img: CaseX },
  { id: 4, title: 'Good Weapon2 1.2', img: CaseX },
  { id: 5, title: 'Good Weapon2 1.2', img: CaseX },
  { id: 6, title: 'Good Weapon2 1.2', img: CaseX }
]

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter keyLength={0}>
        <Case caseItems={caseItems} relaredItems={relaredItems} />
      </MemoryRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
