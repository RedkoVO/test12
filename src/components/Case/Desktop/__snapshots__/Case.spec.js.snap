// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mainWrapperTrade"
>
  <div
    className="Case-containerTradeWrap-1"
  >
    <div
      className="Case-containerTrade-2"
    >
      <header
        className="Case-headerCase-3"
      >
        <ul
          className="MenuNavigation-root-5 dark"
        >
          <li
            className={undefined}
          >
            <div
              className="MenuNavigation-logout-14"
              onClick={[Function]}
            >
              Logout
            </div>
          </li>
          <li>
            <a
              className="MenuNavigation-headerMenuItem-6"
              href="/"
              onClick={[Function]}
            >
              <i
                className="MenuNavigation-menuIcon-7 MenuNavigation-dashboard-8"
              />
              <br />
               Dashboard
            </a>
          </li>
          <li>
            <a
              className="MenuNavigation-headerMenuItem-6"
              href="/"
              onClick={[Function]}
            >
              <i
                className="MenuNavigation-menuIcon-7 MenuNavigation-walletIcon-9"
              />
              <br />
               Wallet
            </a>
          </li>
          <li>
            <a
              className="MenuNavigation-headerMenuItem-6"
              href="/"
              onClick={[Function]}
            >
              <i
                className="MenuNavigation-menuIcon-7 MenuNavigation-shop-10"
              />
              <br />
               Shop
            </a>
          </li>
          <li>
            <a
              className="MenuNavigation-headerMenuItem-6"
              href="/games"
              onClick={[Function]}
            >
              <i
                className="MenuNavigation-menuIcon-7 MenuNavigation-games-11"
              />
              <br />
               Games
            </a>
          </li>
          <li>
            <a
              className="MenuNavigation-headerMenuItem-6"
              href="/"
              onClick={[Function]}
            >
              <i
                className="MenuNavigation-menuIcon-7 MenuNavigation-friends-12"
              />
              <br />
               Friends
            </a>
          </li>
          <li>
            <a
              className="MenuNavigation-headerMenuItem-6"
              href="/"
              onClick={[Function]}
            >
              <i
                className="MenuNavigation-menuIcon-7 MenuNavigation-settings-13"
              />
              <br />
               Settings
            </a>
          </li>
        </ul>
      </header>
      <main
        className="Case-case-4"
      >
        <div
          className="CaseFilter-root-15"
        >
          <input
            className="CaseFilter-caseFilterSearch-16"
            placeholder="Search"
            type="text"
          />
          <div
            className="CaseFilter-caseFilterOptions-17"
          >
            <div
              className="CaseFilter-caseFilterType-18"
            >
              <div
                className="CaseFilter-caseFilterItem-19"
              >
                less than $15&lt;
              </div>
              <div
                className="CaseFilter-caseFilterItem-19"
              >
                $15-30
              </div>
              <div
                className="CaseFilter-caseFilterItem-19 CaseFilter-selectedItem-21"
              >
                 $30-60
              </div>
              <div
                className="CaseFilter-caseFilterItem-19"
              >
                $60-90
              </div>
              <div
                className="CaseFilter-caseFilterItem-19"
              >
                less than $15&lt;
              </div>
            </div>
            <div
              className="CaseFilter-caseFilterType-18"
            >
              <div
                className="CaseFilter-caseFilterItem-19"
              >
                CS:GO
              </div>
              <div
                className="CaseFilter-caseFilterItem-19"
              >
                Dota 2
              </div>
              <div
                className="CaseFilter-caseFilterItem-19 CaseFilter-selectedItem-21"
              >
                Starcraft 2
              </div>
              <div
                className="CaseFilter-caseFilterItem-19"
              >
                 Maincraft
              </div>
              <div
                className="CaseFilter-caseFilterItem-19"
              >
                Rainbow Six
              </div>
            </div>
          </div>
          <div
            className="CaseFilter-caseFilterBtn-22"
          />
        </div>
        <div
          className="CaseOpen-root-23"
        >
          <div
            className="CaseOpen-caseOpenTop-24"
          >
            <div
              className="CaseOpen-caseOpenBackBtn-25"
            />
            <div
              className="CaseOpen-caseOpenTopImg-26"
            >
              <img
                alt=""
                src="caseXOpen.png"
              />
            </div>
            <div
              className="CaseOpen-caseOpenTopDescr-27"
            >
              <div
                className="CaseOpen-caseOpenTopMarker-28"
              />
              <div
                className="CaseOpen-caseOpenTopName-29"
              >
                Case: 
                <span>
                  Hello, buddy!
                </span>
              </div>
              <div
                className="CaseOpen-caseOpenTopPrice-30"
              >
                Open for 
                <span>
                  $34.20
                </span>
              </div>
              <div
                className="CaseOpen-caseOpenTopGame-31"
              >
                Case: 
                <span>
                  CS:GO
                </span>
              </div>
            </div>
          </div>
          <div
            className="CaseOpen-caseOpenContent-32"
          >
            <div
              className="CaseOpen-caseOpenTitle-33"
            >
              This case cointains this stuff:
            </div>
            <div
              className="CaseOpen-caseOpenItem-34"
            >
              <div
                className="CaseOpen-caseOpenItemImg-35"
              >
                <img
                  alt=""
                  src="GunCase.png"
                />
              </div>
              <div
                className="CaseOpen-caseOpenItemName-36"
              >
                Gun: 
                <span>
                  Good Weapon2 1.2
                </span>
              </div>
            </div>
            <div
              className="CaseOpen-caseOpenItem-34"
            >
              <div
                className="CaseOpen-caseOpenItemImg-35"
              >
                <img
                  alt=""
                  src="GunCase.png"
                />
              </div>
              <div
                className="CaseOpen-caseOpenItemName-36"
              >
                Gun: 
                <span>
                  Good Weapon2 1.2
                </span>
              </div>
            </div>
            <div
              className="CaseOpen-caseOpenItem-34"
            >
              <div
                className="CaseOpen-caseOpenItemImg-35"
              >
                <img
                  alt=""
                  src="GunCase.png"
                />
              </div>
              <div
                className="CaseOpen-caseOpenItemName-36"
              >
                Gun: 
                <span>
                  Good Weapon2 1.2
                </span>
              </div>
            </div>
            <div
              className="CaseOpen-caseOpenItem-34"
            >
              <div
                className="CaseOpen-caseOpenItemImg-35"
              >
                <img
                  alt=""
                  src="GunCase.png"
                />
              </div>
              <div
                className="CaseOpen-caseOpenItemName-36"
              >
                Gun: 
                <span>
                  Good Weapon2 1.2
                </span>
              </div>
            </div>
            <div
              className="CaseOpen-caseOpenItem-34"
            >
              <div
                className="CaseOpen-caseOpenItemImg-35"
              >
                <img
                  alt=""
                  src="GunCase.png"
                />
              </div>
              <div
                className="CaseOpen-caseOpenItemName-36"
              >
                Gun: 
                <span>
                  Good Weapon2 1.2
                </span>
              </div>
            </div>
            <div
              className="CaseOpen-caseOpenItem-34"
            >
              <div
                className="CaseOpen-caseOpenItemImg-35"
              >
                <img
                  alt=""
                  src="GunCase.png"
                />
              </div>
              <div
                className="CaseOpen-caseOpenItemName-36"
              >
                Gun: 
                <span>
                  Good Weapon2 1.2
                </span>
              </div>
            </div>
            <div
              className="CaseOpen-caseOpenItem-34"
            >
              <div
                className="CaseOpen-caseOpenItemImg-35"
              >
                <img
                  alt=""
                  src="GunCase.png"
                />
              </div>
              <div
                className="CaseOpen-caseOpenItemName-36"
              >
                Gun: 
                <span>
                  Good Weapon2 1.2
                </span>
              </div>
            </div>
            <div
              className="CaseOpen-caseOpenItem-34"
            >
              <div
                className="CaseOpen-caseOpenItemImg-35"
              >
                <img
                  alt=""
                  src="GunCase.png"
                />
              </div>
              <div
                className="CaseOpen-caseOpenItemName-36"
              >
                Gun: 
                <span>
                  Good Weapon2 1.2
                </span>
              </div>
            </div>
            <div
              className="CaseOpen-caseOpenItem-34"
            >
              <div
                className="CaseOpen-caseOpenItemImg-35"
              >
                <img
                  alt=""
                  src="GunCase.png"
                />
              </div>
              <div
                className="CaseOpen-caseOpenItemName-36"
              >
                Gun: 
                <span>
                  Good Weapon2 1.2
                </span>
              </div>
            </div>
            <div
              className="CaseOpen-caseOpenItem-34"
            >
              <div
                className="CaseOpen-caseOpenItemImg-35"
              >
                <img
                  alt=""
                  src="GunCase.png"
                />
              </div>
              <div
                className="CaseOpen-caseOpenItemName-36"
              >
                Gun: 
                <span>
                  Good Weapon2 1.2
                </span>
              </div>
            </div>
            <div
              className="CaseOpen-caseOpenItem-34"
            >
              <div
                className="CaseOpen-caseOpenItemImg-35"
              >
                <img
                  alt=""
                  src="GunCase.png"
                />
              </div>
              <div
                className="CaseOpen-caseOpenItemName-36"
              >
                Gun: 
                <span>
                  Good Weapon2 1.2
                </span>
              </div>
            </div>
            <div
              className="CaseOpen-caseOpenItem-34"
            >
              <div
                className="CaseOpen-caseOpenItemImg-35"
              >
                <img
                  alt=""
                  src="GunCase.png"
                />
              </div>
              <div
                className="CaseOpen-caseOpenItemName-36"
              >
                Gun: 
                <span>
                  Good Weapon2 1.2
                </span>
              </div>
            </div>
            <div
              className="CaseOpen-caseOpenItem-34"
            >
              <div
                className="CaseOpen-caseOpenItemImg-35"
              >
                <img
                  alt=""
                  src="GunCase.png"
                />
              </div>
              <div
                className="CaseOpen-caseOpenItemName-36"
              >
                Gun: 
                <span>
                  Good Weapon2 1.2
                </span>
              </div>
            </div>
            <div
              className="CaseOpen-caseOpenItem-34"
            >
              <div
                className="CaseOpen-caseOpenItemImg-35"
              >
                <img
                  alt=""
                  src="GunCase.png"
                />
              </div>
              <div
                className="CaseOpen-caseOpenItemName-36"
              >
                Gun: 
                <span>
                  Good Weapon2 1.2
                </span>
              </div>
            </div>
            <div
              className="CaseOpen-caseOpenItem-34"
            >
              <div
                className="CaseOpen-caseOpenItemImg-35"
              >
                <img
                  alt=""
                  src="GunCase.png"
                />
              </div>
              <div
                className="CaseOpen-caseOpenItemName-36"
              >
                Gun: 
                <span>
                  Good Weapon2 1.2
                </span>
              </div>
            </div>
            <div
              className="CaseOpen-caseOpenItem-34"
            >
              <div
                className="CaseOpen-caseOpenItemImg-35"
              >
                <img
                  alt=""
                  src="GunCase.png"
                />
              </div>
              <div
                className="CaseOpen-caseOpenItemName-36"
              >
                Gun: 
                <span>
                  Good Weapon2 1.2
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="CasesRelated-root-37"
        >
          <div
            className="CasesRelated-casesRelatedTilte-38"
          >
            Related to this case:
          </div>
          <div
            className="CasesRelated-casesRelatedItem-39"
          >
            <div
              className="CasesRelated-casesRelatedItemImg-40"
            >
              <img
                alt=""
                src="CaseX.png"
              />
            </div>
            <div
              className="CasesRelated-casesRelatedItemName-41"
            >
              Gun: 
              <span>
                Good Weapon2 1.2
              </span>
            </div>
          </div>
          <div
            className="CasesRelated-casesRelatedItem-39"
          >
            <div
              className="CasesRelated-casesRelatedItemImg-40"
            >
              <img
                alt=""
                src="CaseX.png"
              />
            </div>
            <div
              className="CasesRelated-casesRelatedItemName-41"
            >
              Gun: 
              <span>
                Good Weapon2 1.2
              </span>
            </div>
          </div>
          <div
            className="CasesRelated-casesRelatedItem-39"
          >
            <div
              className="CasesRelated-casesRelatedItemImg-40"
            >
              <img
                alt=""
                src="CaseX.png"
              />
            </div>
            <div
              className="CasesRelated-casesRelatedItemName-41"
            >
              Gun: 
              <span>
                Good Weapon2 1.2
              </span>
            </div>
          </div>
          <div
            className="CasesRelated-casesRelatedItem-39"
          >
            <div
              className="CasesRelated-casesRelatedItemImg-40"
            >
              <img
                alt=""
                src="CaseX.png"
              />
            </div>
            <div
              className="CasesRelated-casesRelatedItemName-41"
            >
              Gun: 
              <span>
                Good Weapon2 1.2
              </span>
            </div>
          </div>
          <div
            className="CasesRelated-casesRelatedItem-39"
          >
            <div
              className="CasesRelated-casesRelatedItemImg-40"
            >
              <img
                alt=""
                src="CaseX.png"
              />
            </div>
            <div
              className="CasesRelated-casesRelatedItemName-41"
            >
              Gun: 
              <span>
                Good Weapon2 1.2
              </span>
            </div>
          </div>
          <div
            className="CasesRelated-casesRelatedItem-39"
          >
            <div
              className="CasesRelated-casesRelatedItemImg-40"
            >
              <img
                alt=""
                src="CaseX.png"
              />
            </div>
            <div
              className="CasesRelated-casesRelatedItemName-41"
            >
              Gun: 
              <span>
                Good Weapon2 1.2
              </span>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
  <footer
    className="Footer-root-42"
  >
    <ul
      className="Footer-footerNav-43"
    >
      <li>
        <a
          className="Footer-footerNavItem-44"
          href="/"
          onClick={[Function]}
        >
          Dashboard
        </a>
      </li>
      <li>
        <a
          className="Footer-footerNavItem-44"
          href="/"
          onClick={[Function]}
        >
          Wallet
        </a>
      </li>
      <li>
        <a
          className="Footer-footerNavItem-44"
          href="/"
          onClick={[Function]}
        >
          Shop
        </a>
      </li>
      <li>
        <a
          className="Footer-footerNavItem-44"
          href="/games"
          onClick={[Function]}
        >
          Games
        </a>
      </li>
      <li>
        <a
          className="Footer-footerNavItem-44"
          href="/"
          onClick={[Function]}
        >
          Friends
        </a>
      </li>
      <li>
        <a
          className="Footer-footerNavItem-44"
          href="/"
          onClick={[Function]}
        >
          Settings
        </a>
      </li>
    </ul>
    <span
      className={undefined}
    >
      Vault. All Rights Reserved 2018
    </span>
  </footer>
</div>
`;
